src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts

```diff
@@ -1,33 +1,406 @@
+
+    responseVisibleSetting: ResponseVisibleSetting.CUSTOM,
+    customResponseVisibleTime: { hour: 23, minute: 59 },
+    customResponseVisibleDate: { year: 0, month: 0, day: 0 },
+
+    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+    publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
+
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+
+    templateSessionName: '',
+
+    isSaving: false,
+    isEditable: false,
+    isDeleting: false,
+    isCopying: false,
+    hasVisibleSettingsPanelExpanded: false,
+    hasEmailSettingsPanelExpanded: false,
+  };
+
   let component: InstructorSessionEditPageComponent;
   let fixture: ComponentFixture<InstructorSessionEditPageComponent>;
+  let feedbackSessionsService: FeedbackSessionsService;
+  let feedbackQuestionsService: FeedbackQuestionsService;
+  let courseService: CourseService;
+  let studentService: StudentService;
+  let instructorService: InstructorService;
+  let navigationService: NavigationService;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [
+        NgbModule,
+        FormsModule,
+        AjaxLoadingModule,
         RouterTestingModule,
         TeammatesRouterModule,
         HttpClientTestingModule,
         InstructorSessionEditPageModule,
+        BrowserAnimationsModule,
       ],
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(InstructorSessionEditPageComponent);
+    courseService = TestBed.inject(CourseService);
+    feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
+    feedbackQuestionsService = TestBed.inject(FeedbackQuestionsService);
+    studentService = TestBed.inject(StudentService);
+    instructorService = TestBed.inject(InstructorService);
+    navigationService = TestBed.inject(NavigationService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with feedback session questions', () => {
+    component.questionEditFormModels = [testQuestionEditFormModel1, testQuestionEditFormModel2];
+    component.isLoadingFeedbackSession = false;
+    component.isLoadingFeedbackQuestions = false;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with new question added', () => {
+    component.isAddingQuestionPanelExpanded = true;
+    component.isLoadingFeedbackSession = false;
+    component.isLoadingFeedbackQuestions = false;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when feedback session failed to load', () => {
+    component.hasLoadingFeedbackSessionFailed = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
```

Comment:
Snapshots with other boolean variables would be welcome too, e.g. is copying question, or whether loading the feedback questions has failed